---
title: "EJP-Soil Fungi data analyse"
author: "Sam Lambrechts", "Silke De Backer"
date: "`r Sys.Date()`"
output:
  bookdown::html_document2:
    toc: true
    toc_float: true
    code_folding: hide
editor_options:
  markdown:
    wrap: sentence
---

```{r setup, include=FALSE}
library(knitr)
library(here)
opts_chunk$set(echo = TRUE, out.width = "100%")
opts_knit$set(root.dir = here::here())

if (!"phyloseq" %in% rownames(installed.packages())) {
  remotes::install_github("joey711/phyloseq")
}
library(phyloseq)
library(vegan)
library(ggplot2)
library(ggrepel)
library(ggpubr)
library(gridExtra)
library(ggbiplot)
library(factoextra)
library(dplyr)
library(rstatix)
library(dunn.test)
library(compositions)
library(sf)
#### Handig pakket met veel phyloseq add-ons
if (!"psadd" %in% rownames(installed.packages())) {
  remotes::install_github("cpauvert/psadd")
}

library(psadd)
if (!"tidytacos" %in% rownames(installed.packages())) {
  remotes::install_github("lebeerlab/tidytacos")
}
if (!"ggVennDiagram" %in% rownames(installed.packages())) {
  install.packages("ggVennDiagram")
}

library(tidytacos)
library(glmmTMB)
library(marginaleffects)
library(performance)

ejp_folder <- "G:/Gedeelde drives/PRJ_EJP-Soil" # nolint
```

# Inlezen data

<!--
In plaats van een .Rdata object, zouden we hier beter de bronbestanden inlezen met de code die hiervoor nodig.
Met bronbestanden bedoel ik hier gewoon de drie tabelletjes die samen het phyloseq object vormen.
Deze tabelletjes kunnen best gewoon csv bestanden zijn die we inlezen.
-->

```{r load-rdata-files}

path_naar_bestand <- file.path(
    ejp_folder,
    "Bodem_biodiversiteit",
    "03_resultaten",
    "phyloseq",
    "Fungi",
    "SRS",
    "met_otuxx",
    "physeq_SRS_Fungi_fungi_species.Rdata"
  )

load(path_naar_bestand)

SRS_fungi_species <- physeq_fungi_fungi.species

rm(physeq_fungi_fungi.species)

path_naar_bestand2 <- file.path(
    ejp_folder,
    "Bodem_biodiversiteit",
    "03_resultaten",
    "phyloseq",
    "Fungi",
    "met_otuxx",
    "physeq_fungi_fungi.Rdata"
  )

load(path_naar_bestand2)

path_naar_bestand3 <- file.path(
    ejp_folder,
    "Bodem_biodiversiteit",
    "03_resultaten",
    "phyloseq",
    "Fungi",
    "met_otuxx",
    "physeq_fungi_fungi_species.Rdata"
  )

load(path_naar_bestand3)
```

Het bestand dat wordt ingelezen is `r basename(path_naar_bestand)` en `r basename(path_naar_bestand2)` en `r basename(path_naar_bestand3)`.

# Preprocessing stappen, Filter en Selectie

verwijder Gontrode, strooiselstalen, en replicaten

```{r hernoem-objecten-en-filter-ongewenste-stalen-en-otus}

# rename physeq objects and remove samples from Gontrode (different project) en litter layer

physeq_srs <- SRS_fungi_species
physeq_nonrarefied <- physeq_fungi_fungi
physeq_nonrarefied_merged <- physeq_fungi_fungi.species

physeq_srs <- subset_samples_no_zero(physeq_srs, Location != "Gontrode")
physeq_nonrarefied <- subset_samples_no_zero(physeq_nonrarefied, Location != "Gontrode")
physeq_nonrarefied_merged <- subset_samples_no_zero(physeq_nonrarefied_merged, Location != "Gontrode")

physeq_srs <- subset_samples_no_zero(physeq_srs, Protocol2 != "litter")
physeq_nonrarefied <- subset_samples_no_zero(physeq_nonrarefied, Protocol2 != "litter")
physeq_nonrarefied_merged <- subset_samples_no_zero(physeq_nonrarefied_merged, Protocol2 != "litter")

# remove singetons from physeq srs

taxa_sums <- taxa_sums(physeq_srs)

singletons <- names(taxa_sums[taxa_sums == 1])

empty_otus <- names(taxa_sums[taxa_sums == 0])

physeq_srs <- prune_taxa(!(taxa_names(physeq_srs) %in% singletons), physeq_srs)

# remove singletons from physeq nonrarefied

taxa_sums <- taxa_sums(physeq_nonrarefied)

singletons <- names(taxa_sums[taxa_sums == 1])

empty_otus <- names(taxa_sums[taxa_sums == 0])

physeq_nonrarefied <- prune_taxa(!(taxa_names(physeq_nonrarefied) %in% singletons), physeq_nonrarefied)

# remove singetons from physeq nonrarefied merged

taxa_sums <- taxa_sums(physeq_nonrarefied_merged)

singletons <- names(taxa_sums[taxa_sums == 1])

empty_otus <- names(taxa_sums[taxa_sums == 0])

physeq_nonrarefied_merged <- prune_taxa(!(taxa_names(physeq_nonrarefied_merged) %in% singletons), physeq_nonrarefied_merged)

# create phyloseq objects with only known taxa

physeq_known_species <- subset_taxa(physeq_nonrarefied_merged, !grepl("_otu", tax_table(physeq_nonrarefied_merged)[, "Species"]))
physeq_known_genera <- subset_taxa(physeq_nonrarefied_merged, !grepl("_otu", tax_table(physeq_nonrarefied_merged)[, "Genus"]))
physeq_known_order <- subset_taxa(physeq_nonrarefied_merged, !grepl("_otu", tax_table(physeq_nonrarefied_merged)[, "Order"]))
physeq_known_family <- subset_taxa(physeq_nonrarefied_merged, !grepl("_otu", tax_table(physeq_nonrarefied_merged)[, "Family"]))
physeq_known_class <- subset_taxa(physeq_nonrarefied_merged, !grepl("_otu", tax_table(physeq_nonrarefied_merged)[, "Class"]))

physeq_srs_known_species <- subset_taxa(physeq_srs, !grepl("_otu", tax_table(physeq_srs)[, "Species"]))
physeq_srs_known_genera <- subset_taxa(physeq_srs, !grepl("_otu", tax_table(physeq_srs)[, "Genus"]))
physeq_srs_known_order <- subset_taxa(physeq_srs, !grepl("_otu", tax_table(physeq_srs)[, "Order"]))
physeq_srs_known_family <- subset_taxa(physeq_srs, !grepl("_otu", tax_table(physeq_srs)[, "Family"]))
physeq_srs_known_class <- subset_taxa(physeq_srs, !grepl("_otu", tax_table(physeq_srs)[, "Class"]))

```

```{r sample-sums-physeq-nonrarefied}
sample_sums(physeq_nonrarefied)
```

```{r sample-sums-physeq-nonrarefied-merged}
sample_sums(physeq_nonrarefied_merged)
```

```{r sample-sums-physeq-srs}
sample_sums(physeq_srs)
```

```{r test}


```


## Conversie voor gebruik in andere packages

Voorbereidende stap: functies om een `phyloseq` object te converteren naar een `vegan matrix` of een `tidytacos` object.


```{r physeq-to-vegan-physeq-srs}
psotu2veg <- function(physeq_srs) {
  otu <- phyloseq::otu_table(physeq_srs)
  if (phyloseq::taxa_are_rows(otu)) {
    otu <- t(otu)
  }
  return(as(otu, "matrix"))
}

veganobject_physeq_srs <- psotu2veg(physeq_srs)

```


```{r physeq-to-vegan-physeq-nonrarefied}
psotu2veg <- function(physeq_nonrarefied) {
  otu <- phyloseq::otu_table(physeq_nonrarefied)
  if (phyloseq::taxa_are_rows(otu)) {
    otu <- t(otu)
  }
  return(as(otu, "matrix"))
}

veganobject_physeq_nonrarefied <- psotu2veg(physeq_nonrarefied)
```

```{r physeq-to-vegan-physeq-nonrarefied-merged}
psotu2veg <- function(physeq_nonrarefied_merged) {
  otu <- phyloseq::otu_table(physeq_nonrarefied_merged)
  if (phyloseq::taxa_are_rows(otu)) {
    otu <- t(otu)
  }
  return(as(otu, "matrix"))
}

veganobject_physeq_nonrarefied_merged <- psotu2veg(physeq_nonrarefied_merged)
```



```{r convert-tidytacos}
tidy_physeq_srs <- tidytacos::from_phyloseq(physeq_srs)

tidy_physeq_nonrarefied <- tidytacos::from_phyloseq(physeq_nonrarefied)

tidy_physeq_nonrarefied_merged <- tidytacos::from_phyloseq(physeq_nonrarefied_merged)

tidy_physeq_known_species <- tidytacos::from_phyloseq(physeq_known_species)

tidy_physeq_known_genera <- tidytacos::from_phyloseq(physeq_known_genera)

tidy_physeq_known_family <- tidytacos::from_phyloseq(physeq_known_family)

tidy_physeq_known_order <- tidytacos::from_phyloseq(physeq_known_order)

tidy_physeq_known_class <- tidytacos::from_phyloseq(physeq_known_class)


tidy_physeq_srs_known_species <- tidytacos::from_phyloseq(physeq_srs_known_species)

tidy_physeq_srs_known_genera <- tidytacos::from_phyloseq(physeq_srs_known_genera)

tidy_physeq_srs_known_family <- tidytacos::from_phyloseq(physeq_srs_known_family)

tidy_physeq_srs_known_order <- tidytacos::from_phyloseq(physeq_srs_known_order)

tidy_physeq_srs_known_class <- tidytacos::from_phyloseq(physeq_srs_known_class)


tidy_physeq_srs <- tidy_physeq_srs %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()


tidy_physeq_nonrarefied <- tidy_physeq_nonrarefied %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()

tidy_physeq_nonrarefied_merged <- tidy_physeq_nonrarefied_merged %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()


tidy_physeq_known_species <- tidy_physeq_known_species %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()

tidy_physeq_known_genera <- tidy_physeq_known_genera %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()

tidy_physeq_known_family <- tidy_physeq_known_family %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()

tidy_physeq_known_order <- tidy_physeq_known_order %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()

tidy_physeq_known_class <- tidy_physeq_known_class %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()



tidy_physeq_srs_known_species <- tidy_physeq_srs_known_species %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()

tidy_physeq_srs_known_genera <- tidy_physeq_srs_known_genera %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()

tidy_physeq_srs_known_family <- tidy_physeq_srs_known_family %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()

tidy_physeq_srs_known_order <- tidy_physeq_srs_known_order %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()

tidy_physeq_srs_known_class <- tidy_physeq_srs_known_class %>%
  remove_empty_samples() %>%
  tidytacos::set_rank_names(
    rank_names = c("phylum", "class", "order", "family", "genus", "species")
  ) %>%
  tidytacos::add_alpha() %>%
  tidytacos::add_total_count()

```

# Verkennende analyses


```{r}
physeq_nonrarefied
```

```{r}
physeq_nonrarefied_merged
```

```{r}
physeq_srs
```

```{r}
physeq_known_species
```

```{r}
physeq_known_genera
```

```{r}
physeq_known_family
```

```{r}
physeq_known_order
```

```{r}
physeq_known_class
```

```{r}
physeq_srs_known_species
```

```{r}
physeq_srs_known_genera
```

```{r}
physeq_srs_known_family
```

```{r}
physeq_srs_known_order
```

```{r}
physeq_srs_known_class
```

## Sample data

```{r verken-sample-data}
glimpse(sample_data(physeq_srs) %>% as_tibble())
```

Check totaal aantal reads (hier de decielen van 0% = minimum tot 100% = maximum):

```{r quantiles-reads}
sample_sums(physeq_nonrarefied) %>%
  quantile(probs = seq(0, 1, 0.1))
```

```{r quantiles-reads-species-merged}
sample_sums(physeq_nonrarefied_merged) %>%
  quantile(probs = seq(0, 1, 0.1))
```


We verkennen ook de verdeling van de samples over de belangrijkste design-variabelen:

```{r designvars}
tidy_physeq_srs$samples %>%
  count(Protocol2, Land_use) %>%
  kable()
```

Aantal bodemlocaties:

```{r}
tidy_physeq_srs$samples %>%
  group_by(Land_use) %>%
  summarise(n_locaties = n_distinct(Name)) %>%
  kable()

```

```{r}
tidy_physeq_srs$samples %>%
  group_by(Protocol2) %>%
  summarise(n_locaties = n_distinct(Name)) %>%
  kable()
```

De meeste 0-10 cm en 10-30 cm stalen zijn van dezelfde locatie.
Met deze gepaardheid moeten we rekening houden in analyses.


## OTU tabel

```{r verken-otu-data}


glimpse(otu_table(physeq_srs) %>% as.data.frame %>% as_tibble())
```

## Taxonomie tabel


```{r verken-taxonomie-data}
glimpse(tax_table(physeq_srs) %>% as.data.frame %>% as_tibble())
```

```{r GBIF-check-presence}

Species <- as.vector(tax_table(physeq_known_species)[, "Species"])
Species <- Species[!grepl("unidentified|\\?", Species)]

Species <- gsub("_", " ", Species)
source(here::here("source/r/check_presence.R"))
gbif_check <- check_presence(Species)
gbif_check %>%
filter(!present) %>%
  kable(caption = "Not present according to GBIF in Western-Europe")
```


## Unieke en gedeelde taxa per staalname protocol per locatie:


```{r detectie-otus-staalname-protocol}

# Your original phyloseq object (replace with your actual object)
phyloseq_venn <- physeq_srs

print(phyloseq_venn)

# Extract sample data
sample_data <- as.data.frame(sample_data(phyloseq_venn))

# Extract unique locations and corresponding protocols
locations <- unique(sample_data$Location)
protocols <- lapply(locations, function(loc) unique(sample_data$Protocol[sample_data$Location == loc]))

# Initialize vector to store samples to be removed
samples_to_remove <- character(0)

# Identify samples to be removed based on conditions
for (i in seq_along(locations)) {
  if (length(protocols[[i]]) == 1 && protocols[[i]] == "INBO") {
    samples_to_remove <- c(samples_to_remove, rownames(sample_data[sample_data$Location == locations[i], ]))
  }
}

# Remove duplicates
samples_to_remove <- unique(samples_to_remove)

# Which samples need to be removed?
samples_to_remove


# Remove samples
phyloseq_venn_filtered <- subset_samples_no_zero(phyloseq_venn, !(Name %in% samples_to_remove))
phyloseq_venn_filtered <- subset_samples_no_zero(phyloseq_venn_filtered, Layer != "litter")


# Make venn_diagram per
# Per locatie
phyloseq_venn_filtered_location <- phyloseq_venn_filtered

# List of unique locations
locations <- unique(sample_data(phyloseq_venn_filtered_location)$Location)

# Create an empty list to store the subsetted phyloseq objects
phyloseq_list <- list()

# Loop through each location
for (loc in locations) {
  # Subset the phyloseq object for the current location
  phyloseq_sub <- subset_samples_no_zero(phyloseq_venn_filtered_location, Location == loc)
  
  # Assign the subsetted phyloseq object to a variable with the location name
  assign(paste0("phyloseq_", loc), phyloseq_sub)
  
  # Store the subsetted phyloseq object in the list
  phyloseq_list[[loc]] <- phyloseq_sub
}


# List to store plots
plots_venn_locations <- list()


# Loop over each location
for (location in locations) {
  # Create tidy_physeq object
  tidy_physeq_venn <- tidytacos::from_phyloseq(get(paste0("phyloseq_", location)))
  
  # Generate plot with specified size and add title
  plot <- tidy_physeq_venn %>%
    tidytacos::tacoplot_venn(Protocol2) +
    ggtitle(paste("Location:", location)) +
    expand_limits(x = c(-8, 10), y = c(-5, 5))  # Adjust expansion here
  
  # Store plot in the list
  plots_venn_locations[[location]] <- plot
}

# Now you can access each plot from the list and display it
for (location in locations) {
  print(plots_venn_locations[[location]])
}


```
### Gekende soorten al dan niet gemist met bepaalde staalname protocollen per locatie
#### INBO_0_10 vs Lucas
```{r Species-detected-with-INBO_0_10vsLUCAS}

# INBO_0_10 vs Lucas


```


```


Unieke en gedeelde taxa per Protocol2 over heel de dataset:

```{r}
tidy_physeq_srs %>%
  tidytacos::tacoplot_venn(Protocol2)
```

Unieke en gedeelde taxa per Land_use:

```{r}
tidy_physeq_srs %>%
  tidytacos::tacoplot_venn(Land_use)
```

```{r}
tidy_physeq_srs %>%
  tidytacos::tacoplot_stack()
```

```{r Krona, eval=FALSE}
# Create a new combined column
combined_column <- paste(sample_data(physeq_rarefied)$Land_use, sample_data(physeq_rarefied)$Protocol2, sep = "_")

# Add the new combined column to the sample metadata with the desired name
sample_data(physeq_rarefied)$Land_use_Protocol2 <- combined_column

# Update the phyloseq object with the modified sample data
physeq_rarefied <- merge_phyloseq(physeq_rarefied, sample_data(physeq_rarefied))


if (system(command = "which ktImportText",  intern = FALSE, ignore.stderr = TRUE,  ignore.stdout = TRUE) != 1) {
  psadd::plot_krona(physeq_rarefied, "ejp_Olig01_fungi_rar_Species_alle_stalen_Land_use_per_Protocol2", "Land_use_Protocol2", trim = T)
}



```


```{r eval=FALSE}
tidy_physeq_srs %>%
  tidytacos::tacoplot_stack(x = Land_use)
```

```{r eval=FALSE}
tidy_physeq_srs %>%
  tidytacos::tacoplot_stack(x = Protocol2)
```


## Alfa-diversiteit

### Regressiemodel met correctie voor totaal aantal reads

Het geobserveerde aantal taxa neemt toe met het aantal reads in een staal:

```{r}
tidy_physeq_nonrarefied_merged$samples %>%
  filter(total_count > 0) %>%
  ggplot(
    aes(x = total_count, y = observed)) +
  geom_point() +
  scale_x_log10() +
  labs(y = "Observed number of taxa")
```

We fitten een model waarbij we de log van het totaal aantal reads in een staal als covariaat toevoegen samen met Land_use en Protocol2 van het staal (de laatste twee in interactie).
We voegen ook een random intercept toe die aangeeft op welke locatie een staal werd genomen.
Hiermee geven we aan dat stalen op verschillende Protocol2, maar dezelfde locatie, gecorreleerd zijn.
Voor deze analyse, verwijderen we enkele stalen met een laag totaal aantal reads ($\leq 1e+4$).

We modelleren de soortenrijkdom als een Negatief Binomiaal model omdat we verwachten dat deze data overdispersie vertonen ten opzichte van een Poisson verdeling (waar de veronderstelling is dat de variantie gelijk aan het gemiddelde).

```{r m-fungi-richness, eval=FALSE}
samples_totcount_10k <- tidy_physeq$samples %>%
  filter(total_count > 1e+4)

form_fungi_richness <- formula(
  observed ~
    log(total_count)
  + Land_use
  + Protocol2
  + Land_use:Protocol2
  + (1 | Locatie)
  )

m_fungi_richness <- glmmTMB::glmmTMB(
  formula = form_fungi_richness,
  data = samples_totcount_10k,
  family = glmmTMB::nbinom2())
```

Model validatie toont dat er problemen zijn met de residuele variabiliteit.
Er lijkt vooral overdispersie (variabiliteit groter dan verwacht) te zijn bij grote waarden van de soortenrijkdom.


```{r eval=FALSE}
performance::check_model(m_fungi_richness)
```


```{r m-fungi-richness-summary, eval=FALSE}
summary(m_fungi_richness)
```

```{r m-fungi-richness-anova, eval=FALSE}
car::Anova(m_fungi_richness)
```

Zoals te verwachten, is het belangrijk om te corrigeren voor het totaal aantal reads:

```{r eval=FALSE}
marginaleffects::plot_predictions(
  m_fungi_richness,
  condition = c("total_count"),
  vcov = vcov(m_fungi_richness)) +
  labs(y = "Voorspeld aantal taxa") +
  scale_x_log10()
```


We kunnen nu predicties maken van soortenrijkdom waarbij we controleren voor het totaal aantal reads in een staal:

```{r m-fungi-richness-predictions, eval=FALSE}
marginaleffects::plot_predictions(
  m_fungi_richness,
  condition = c("Land_use", "Protocol2"),
  vcov = vcov(m_fungi_richness),
  re.form = NA,
  type = "response") +
  labs(y = "Voorspelde aantal taxa")
```


### Rarefaction analyse 

Dit is een alternatief voor voorgaande analyse.
Schatting van de soortenrijkdom via rarefaction analyse.

```{r rarefaction}
sam_new <- data.frame(sample_data(physeq_nonrarefied_merged))


estimated_Species_richness <- vegan::rarefy(
  veganobject_physeq_nonrarefied_merged,
  min(sample_sums(physeq_nonrarefied_merged)),
  se = TRUE,
  MARGIN = 1)


true_rarefaction_results <- cbind(
  sam_new %>% select(Name, Protocol2, Land_use),
  est = estimated_Species_richness["S", ],
  se = estimated_Species_richness["se", ]) %>%
  as_tibble() %>%
  mutate(est_lower = est - 2 * se,
         est_upper = est + 2 * se)
```


```{r}
glimpse(true_rarefaction_results)
# verwijder gevallen waar se te klein is of nul is
true_rarefaction_results <- true_rarefaction_results %>%
  filter(se > 0.1)
```

```{r m-rarefied}
library(brms)

form_rarefaction <- bf(
  est | se(se, sigma = TRUE) ~
  + Land_use
  + Protocol2
  + Land_use:Protocol2
  + (1 | Cmon_PlotID)
)

m_rarefaction <- brm(
  formula = form_rarefaction,
  data = true_rarefaction_results,
  family = gaussian())

```



```{r m-rarefied-summary}
summary(m_rarefaction)
```

Dit model geeft gelijkaardige resultaten als het model waarbij we rechtstreeks werkten met de geobserveerde aantallen taxa en (de log van) het totaal aantal reads als covariaat toevoegden.

```{r m-rarefied-predictions}
conditional_effects(
  m_rarefaction,
    effects = c("Land_use:Protocol2"),
  re.form = NA)
```

## OTU counts 

per phylum

```{r per-phylum}
# Count OTUs per Phylum
tax_table(physeq) %>%
  as.data.frame() %>%
  as_tibble() %>%
  group_by(phylum) %>%
  summarise(OTU_Count = n()) %>%
  kable()
```


per class

```{r per-class}
# Count OTUs per Class
tax_table(physeq) %>%
  as.data.frame() %>%
  as_tibble() %>%
  group_by(class) %>%
  summarise(OTU_Count = n()) %>%
  kable()
```


per order

```{r per-order}
# Count OTUs per order
tax_table(physeq) %>%
  as.data.frame() %>%
  as_tibble() %>%
  group_by(order) %>%
  summarise(OTU_Count = n()) %>%
  kable()
```

per family

```{r per-family}
# Count OTUs per family
tax_table(physeq) %>%
  as.data.frame() %>%
  as_tibble() %>%
  group_by(family) %>%
  summarise(OTU_Count = n()) %>%
  kable()
```


Per genus

```{r per-genus}
tax_table(physeq) %>%
  as.data.frame() %>%
  group_by(genus) %>%
  summarise(OTU_Count = n()) %>%
  arrange(desc(OTU_Count)) %>%
  kable()
```

Per Species (check of er Species zijn met meer dan 1 OTU)

```{r per-Species}
tax_table(physeq) %>%
  as.data.frame() %>%
  group_by(Species) %>%
  summarise(OTU_Count = n()) %>%
  filter(OTU_Count > 1) %>%
  kable()
```


## Read counts

Overzicht van som van reads per phylum:

```{r}
tidy_physeq %>%
  tidytacos::aggregate_taxa(rank = "phylum") %>%
  tidytacos::everything() %>%
  group_by(phylum) %>%
  summarise(read_counts = sum(count)) %>%
  arrange(desc(read_counts)) %>%
  kable()
```

Overzicht van som van reads per class:

```{r}
tidy_physeq %>%
  tidytacos::aggregate_taxa(rank = "class") %>%
  tidytacos::everything() %>%
  group_by(class) %>%
  summarise(read_counts = sum(count)) %>%
  arrange(desc(read_counts)) %>%
  kable()
```

Overzicht van som van reads per family:

```{r}
tidy_physeq %>%
  tidytacos::aggregate_taxa(rank = "family") %>%
  tidytacos::everything() %>%
  group_by(family) %>%
  summarise(read_counts = sum(count)) %>%
  arrange(desc(read_counts)) %>%
  kable()
```

Overzicht van som van reads per genus:

```{r}
tidy_physeq %>%
  tidytacos::aggregate_taxa(rank = "genus") %>%
  tidytacos::everything() %>%
  group_by(genus) %>%
  summarise(read_counts = sum(count)) %>%
  arrange(desc(read_counts)) %>%
  kable()
```

Overzicht van som van reads per species:

```{r}
tidy_physeq %>%
  tidytacos::aggregate_taxa(rank = "species") %>%
  tidytacos::everything() %>%
  group_by(species) %>%
  summarise(read_counts = sum(count)) %>%
  arrange(desc(read_counts)) %>%
  kable()
```

## Ordinatie

```{r physec-rarefied}

```

```{r ordination-vegan}
ordmat <- psotu2veg(physeq_rarefied)

all(colSums(ordmat) > 0)
all(rowSums(ordmat) > 0)
all(ordmat >= 0)

ordmat <- decostand(ordmat, method = "rclr")

samples <- sample_data(physeq_rarefied) %>%
  as.data.frame() %>%
  as_tibble()

rda <- rda(ordmat)

biplot(rda)


```


```{r eval=FALSE}
# Perform CLR transformation
clr_physeq_rarefied <- transform(physeq_rarefied, "clr")

# Calculate Bray distance matrix based on the CLR-transformed data
bray_distance <- distance(physeq, method = "bray")

# Perform PCoA
pcoa_result <- ordinate(physeq, method = "PCoA", distance = "bray")

# Create the ordination plot with color based on land_use and shape
# based on Protocol2
plot_ordination(physeq, pcoa_result, type = "samples",
                color = "Land_use", shape = "Protocol2") +
  theme_minimal()
```
